packages/app-runtime/src/modules/runtimeEvents/DatawalletSynchronizedModule.ts

```diff
@@ -33,7 +34,7 @@ export class DatawalletSynchronizedModule extends AppRuntimeModule<DatawalletSyn
 
         switch (mostRecentIdentityDeletionProcess.status) {
             case IdentityDeletionProcessStatus.Approved:
-                newDeletionDate = mostRecentIdentityDeletionProcess.gracePeriodEndsAt;
+                newDeletionDate = CoreDate.from(mostRecentIdentityDeletionProcess.gracePeriodEndsAt!);
```

Comment:
hmm, why do we need the null-check here? And is the null-check safe, b/c it's in status Approved?

Reply:
`gracePeriodEndsAt` is an optional property, that is only set if the status is Approved. We know that, but the compiler doesn't. I suppose we could add type guards for the IdentityDeletionProcess, but I'm not sure if it's worth the effort.

Reply:
Ok, that's all I wanted to know! Great!